components/MintButton/MintButton.tsx

```diff
@@ -1,101 +1,18 @@
-import { FC, useState } from "react"
 import Confetti from "react-confetti"
-import axios from "axios"
-import { useRouter } from "next/router"
 import useWindowSize from "../../lib/useWindowSize"
-import { storeBlob } from "../../lib/ipfs"
-import transformTwitterHandle from "../../lib/transformTwitterHandle"
 
-const textToImage = require("text-to-image")
+import { useNFTNameGenerator } from "../../providers/NFTNameGeneratorProvider"
 
-interface MintButtonProps {
-  twitterHandle?: string
-}
-const MintButton: FC<MintButtonProps> = ({ twitterHandle }) => {
-  const router = useRouter()
-  const [loading, setLoading] = useState(false)
-  const [startConfetti, setStartConfetti] = useState(false)
+const MintButton = () => {
+  const { loading, startConfetti, handleGenerateClick } = useNFTNameGenerator()
   const { width, height } = useWindowSize()
 
```

Comment:
moved all this code to provider that way the mint button component is not doing anything more than displaying a button.

Reply:
great idea 💡 
- Clean Code: single-responsibility-principle